prov/peer/src/peer_cq.c

```diff
@@ -66,21 +66,27 @@ static struct fi_ops_cq coll_cq_ops = {
 	.strerror = fi_no_cq_strerror,
 };
 
-int coll_cq_open(struct fid_domain *domain, struct fi_cq_attr *attr,
-		 struct fid_cq **cq_fid, void *context)
+int peer_cq_init(struct fid_domain *domain, struct fi_cq_attr *attr,
+		struct fid_cq **cq_fid, struct fi_peer_cq_context *peer_context)
 {
-	struct coll_cq *cq;
-	struct fi_peer_cq_context *peer_context = context;
+	struct peer_cq *cq;
 	int ret;
 
+	const struct util_domain *util_domain;
+	const struct fi_provider* provider;
+
+	util_domain = container_of(domain, struct util_domain, domain_fid.fid);
+	provider = util_domain->fabric->prov;
```

Comment:
I would update the call to take the provider pointer as an input parameter, so we don't need to dereference the domain pointer.  That matches how other calls handle this situation (e.g. see ofi_cq_init() below).

Reply:
Done